### DISCUSSION TITLE
Does TheRealWorld from Andrew Tate Infringe the AGPL License of Revolt?

### DISCUSSION THREAD

**Discussion by schettn** [stoatchat/discussions#862] [General]
Hi Revolt Team,

I wanted to bring to your attention a potential license infringement issue regarding TheRealWorld platform, which is associated with Andrew Tate.

**Observation:**
- TheRealWorld uses Revolt as a backend and appears to have potentially modified the frontend source code. For instance, their API can be found at [https://api.therealworld.ag](https://api.therealworld.ag).
- The frontend of TheRealWorld looks quite similar to Revolt's frontend, which suggests that it might be a modified clone of Revolt.

**AGPL License Considerations:**
As you know, Revolt is licensed under the AGPL v3, which requires that any modifications to the source code, especially if they are distributed or used over a network, must be made available under the same AGPL license.

**Questions:**
1. **Agreement with TheRealWorld**: Do you have any agreement or deal with TheRealWorld that allows them to use and modify the Revolt backend and frontend without releasing their modified source code? If so, could you please provide some details on this arrangement?
   
2. **License Compliance**: If there is no such agreement, does TheRealWorld's use and potential modification of Revolt's code constitute a violation of the AGPL license? Given that their platform appears to be using a modified version of Revolt, should they not be obligated to release their modified source code publicly?

3. **Next Steps**: If it is determined that TheRealWorld is infringing the AGPL license, what actions can be taken to address this issue? Are there plans to reach out to them for compliance or any other steps that the community should be aware of?

I believe it is crucial for the integrity of open-source projects to ensure that the licensing terms are respected and adhered to. Clarifying this situation will help maintain the trust and transparency that open-source communities rely on.

Update: @insertish  I just found your article about this: https://insrt.uk/post/andrew-tate-stealing-software-revolt

**Comment by schettn** [comment]
@insertish FYI: A new Youtube video that was just released yesterday mentioned the infringement. Maybe it will get some attention now.

https://youtu.be/mV98gVg36p0?si=ubr-YMVeYknUEaKI&t=2014

**Comment by bloomsirenix** [comment]
there also acourding to NTTS homophobic so please sue them and if you want share the money with people who can make a actual platform that also **respects the AGPL and maybe even connects to revolt instead of a shitstorm like Andrew made** also ask them to release there code I gladly rip out those pesky paywall features and make a revolt but with premium features that are not infringing or money grabs like AI TTS,image gen etc
Contact me on discord(angelfencer) or revolt(Userid: 01FT1YS7X8CE7R0BPWH4P61TNA ) as I had a plan to make something like discord using revolt that is still profitable while being not paid to win and overcommercial

**Comment by lapidary-master** [comment]
I got a little curious and I realised that the [revoltchat/self-hosted](https://github.com/revoltchat/self-hosted/) doesn't have a license, whilst someone may assume that no license means it's as open as anything, that is actually not the case and it means [All Rights Reserved](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository#choosing-the-right-license). But to be honest, on top of that, the are using [revoltchat/autumn](https://github.com/revoltchat/autumn) and [revoltchat/january](https://github.com/revoltchat/january) which are licensed under AGPL so, yeah. Maybe someone like the EFF or a like-minded org would be willing to help the legal and slap that bitch around the courts.

**Comment by schettn** [comment]
@lapidary-master This discussion aged well, they got hacked.

 https://nypost.com/2024/11/22/business/hackers-breach-andrew-tates-online-university-website-steal-user-names-and-chat-logs/
